lib/ovirt/service.rb

```diff
@@ -127,11 +128,13 @@ def status(link)
     end
 
     def api_uri(path = nil)
-      uri = "#{base_uri}/api"
+      uri = "#{base_uri}"
+      uri += api_version == 'v4' ? '/ovirt-engine/api' : '/api'
       unless path.nil?
         parts = path.to_s.split('/')
-        parts.shift if parts.first == ''    # Remove leading slash
-        parts.shift if parts.first == 'api' # We already have /api in our URI
+        parts.shift if parts.first == ''             # Remove leading slash
+        parts.shift if parts.first == 'ovirt-engine' # We already have /ovirt-engine in our URI
+        parts.shift if parts.first == 'api'          # We already have /api in our URI
         uri += "/#{parts.join('/')}" unless parts.empty?
```

Comment:
I think that most users can get the API URL.
Also, it is a one-time effort to configure the provider.

However, I'm okay with other solutions, as long as they will allow to set
it manually, in case it is needed.
Is it possible?

On Mon, Mar 7, 2016 at 11:06 PM, Joe Rafaniello <notifications@github.com>
wrote:

> In lib/ovirt/service.rb
> <https://github.com/ManageIQ/ovirt/pull/47#discussion_r55271987>:
>
> >        unless path.nil?
> >          parts = path.to_s.split('/')
> > -        parts.shift if parts.first == ''    # Remove leading slash
> > -        parts.shift if parts.first == 'api' # We already have /api in our URI
> > +        parts.shift if parts.first == ''             # Remove leading slash
> > +        parts.shift if parts.first == 'ovirt-engine' # We already have /ovirt-engine in our URI
> > +        parts.shift if parts.first == 'api'          # We already have /api in our URI
> >          uri += "/#{parts.join('/')}" unless parts.empty?
>
> Note, we might be fine with ipv6 now as long as we don't build strings
> using the hostname, but URI just handles that for you. I dislike the
> URI#split returning an array of 9 elements but at least you don't have to
> worry about IPv6.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ManageIQ/ovirt/pull/47/files#r55271987>.
>